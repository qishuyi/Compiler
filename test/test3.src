3 +
(if
    3 <=0
 then
    true
 else
    false)