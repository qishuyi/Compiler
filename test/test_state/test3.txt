let x : < [ int ] > = ref [] : int in
  let counter : <int> = ref 0 in
    while !counter <= 5 do x := !counter :: !x ; counter := counter + 1 end ; hd !x
