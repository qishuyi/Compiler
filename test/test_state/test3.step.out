(let x = (ref ([] : int)) in (let counter = (ref 0) in (while (<= !counter 5) do (x := (!counter :: !x)) ; (counter := (+ counter 1)) end) ; (hd !x)))
(let x = Ptr(0) in (let counter = (ref 0) in (while (<= !counter 5) do (x := (!counter :: !x)) ; (counter := (+ counter 1)) end) ; (hd !x)))
(let counter = (ref 0) in (while (<= !counter 5) do (Ptr(0) := (!counter :: !Ptr(0))) ; (counter := (+ counter 1)) end) ; (hd !Ptr(0)))
(let counter = Ptr(1) in (while (<= !counter 5) do (Ptr(0) := (!counter :: !Ptr(0))) ; (counter := (+ counter 1)) end) ; (hd !Ptr(0)))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
(while (<= !Ptr(1) 5) do (Ptr(0) := (!Ptr(1) :: !Ptr(0))) ; (Ptr(1) := (+ Ptr(1) 1)) end) ; (hd !Ptr(0))
() ; (hd !Ptr(0))
() ; (hd (5 :: (4 :: (3 :: (2 :: (1 :: (0 :: ([] : int))))))))
() ; 5
5
